packages/elements/src/internals/machines/sign-in/sign-in.machine.ts

```diff
@@ -141,6 +142,14 @@ export const SignInMachine = setup({
         };
       },
     ),
+    setLoadingContext: assign(({ event }) => {
+      if (event.type === 'SET_LOADING_CONTEXT') {
+        return {
+          loadingContext: event.name,
+        };
+      }
+      return {};
+    }),
```

Comment:
There's likely a more elegant way we can store an identifier for whatever action / element triggered the loading state, this was my first stab at it.

Reply:
This feels best handled when `AUTHENTICATE.OAUTH`/`AUTHENTICATE.[...]` are called, rather than their own events, unless I'm missing something.